tests/integrations/aws_lambda/test_aws.py

```diff
@@ -234,3 +239,50 @@ def test_handler(event, context):
         "query_string": {"bonkers": "true"},
         "url": "https://iwsz2c7uwi.execute-api.us-east-1.amazonaws.com/asd",
     }
+
+
+def test_init_error(run_lambda_function):
+    events, response = run_lambda_function(
+        LAMBDA_PRELUDE
+        + dedent(
+            """
+        init_sdk()
+        func()
+        def test_handler(event, context):
+            return 0
+        """
+        ),
+        b'{"foo": "bar"}',
+    )
+    log_result = (base64.b64decode(response["LogResult"])).decode("utf-8")
+    expected_text = "name 'func' is not defined"
+    assert expected_text in log_result
+
+
+def test_timeout_error(run_lambda_function):
+    # Modifying LAMBDA_PRELUDE since capturing timeout error is kept optional.
+    modified_prelude = LAMBDA_PRELUDE.replace("[AwsLambdaIntegration()]", "[AwsLambdaIntegration(True)]")
+    events, response = run_lambda_function(
+        modified_prelude
+        + dedent(
+            """
+        init_sdk()
+        def test_handler(event, context):
+            time.sleep(10)
+            return 0
+        """
+        ),
+        b'{"foo": "bar"}',
+    )
+    expected_text = "WARNING : Function is expected to get timed out. Configured timeout duration = 4 seconds"
+    if not events:
+        # In case of Python 2.7 runtime environment
```

Comment:
Tests should not rely on the truthy value of `events` and infer it is Python 2.7 or 3.x.
E.g., if something is wrong in Python 3.x and `events` is the empty list, the test would go into the wrong branch of the `if...else`.

Reply:
Also I am confused, does this mean timeout events are not actually working on python 2.7?